gems/smithy-client/lib/smithy-client/protocols/rpc_v2.rb

```diff
@@ -27,7 +27,7 @@ def build(context)
         def parse(context)
           output_shape = context.operation.output
           codec = Client::Codecs::CBOR.new(setting(context))
-          codec.deserialize(context.response.body.read, output_shape, output_shape.type)
+          codec.deserialize(context.response.body.read, output_shape, output_shape.type.new)
```

Comment:
Possibly an issue with codec interface. If output shape is already passed in, why also type? And maybe codec should do that initialization.

Reply:
Lets discuss on protocol implementation PR.

Reply:
Yeah I need to update my PR on this - basically, we want to only pass in type when it comes to typed documents (since it will contain a discriminator id for us to look up in the TypeRegistry). Otherwise, we should just use the type associated with the output_shape.